File: services/payment-service/src/main/java/com/movra/payment/dto/ApiError.java
package com.movra.payment.dto;

import lombok.Builder;
import lombok.Data;

import java.time.Instant;
import java.util.Map;

@Data
@Builder
public class ApiError {
    private String code;
    private String message;
    private Map<String, String> details;
    private Instant timestamp;
    private String path;
}

File: services/auth-service/src/Dtos/ApiError.cs
namespace AuthService.Dtos;

public class ApiError
{
    public string Code { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public Dictionary<string, string>? Details { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    public string? Path { get; set; }
}
